indykite/data_source_application.go

```diff
@@ -72,46 +74,94 @@ func dataSourceApplicationList() *schema.Resource {
 	}
 }
 
+// lookupApplicationByName finds an application by name within an app space.
+func lookupApplicationByName(
+	ctx context.Context,
+	clientCtx *ClientContext,
+	data *schema.ResourceData,
+	name string,
+) (ApplicationResponse, diag.Diagnostic) {
+	// Get project ID from either project_id or app_space_id (backward compatibility)
+	projID := data.Get(projectIDKey).(string)
+	if projID == "" {
+		projID = data.Get(appSpaceIDKey).(string)
+	}
+
+	// Validate that at least one of app_space_id or project_id is provided when using name
+	if projID == "" {
+		return ApplicationResponse{}, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "\"name\": all of `app_space_id, name` must be specified, or use `project_id, name` instead",
+		}
+	}
+
+	var listResp ListApplicationsResponse
+	err := clientCtx.GetClient().Get(ctx, "/applications?project_id="+projID+"&search=", &listResp)
+	if err != nil {
+		return ApplicationResponse{}, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  fmt.Sprintf("failed to list applications: %v", err),
+		}
+	}
+
+	// Find application by name
+	for i := range listResp.Applications {
+		if listResp.Applications[i].Name == name {
+			return listResp.Applications[i], diag.Diagnostic{}
+		}
+	}
```

Comment:
Why you are searching by name? This is supported by REST API

Reply:
updated

Reply:
updated